pytorch_lightning/utilities/auto_restart.py

```diff
@@ -280,6 +281,26 @@ def set_rng_states(rng_state_dict: Dict[str, Any]) -> None:
     python_set_rng_state((version, tuple(state), gauss))
 
 
+@contextmanager
+def isolate_rng() -> Generator[None, None, None]:
```

Comment:
I'm open to other name suggestions of course :) 

The definition is here because also the other functions `collect_rng_states` and `set_rng_states` are defined here. Another option is to move them to the utilities/seed.py file but it's also not 100% fitting the semantics.

Reply:
How about `isolate_randomize` as you can also run shuffle inside, right?

Reply:
I like `isolate_rng`.

But I would move it to `seed.py`, to me, `auto_restart` is for fault-tolerance support